File: src/main/java/com/lcaohoanq/nocket/domain/token/TokenRepository.java
package com.lcaohoanq.nocket.domain.token;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.transaction.annotation.Transactional;

public interface TokenRepository extends JpaRepository<Token, UUID> {

    List<Token> findByUserId(UUID user_id);

    Token findByToken(String token);

    Token findByRefreshToken(String token);

    @Modifying
    @Transactional
    @Query("DELETE FROM Token t WHERE t.expirationDate < :now OR t.expired = true")
    void deleteExpiredTokens(@Param("now") LocalDateTime now);


}

File: src/main/java/com/lcaohoanq/nocket/domain/token/TokenRepository.kt
package com.lcaohoanq.nocket.domain.token

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Modifying
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import org.springframework.transaction.annotation.Transactional
import java.time.LocalDateTime
import java.util.*

interface TokenRepository : JpaRepository<Token, UUID?> {
    fun findByUserId(userId: UUID?): List<Token>

    fun findByToken(token: String?): Token

    fun findByRefreshToken(token: String?): Token?

    @Modifying
    @Transactional
    @Query("DELETE FROM Token t WHERE t.expirationDate < :now OR t.expired = true")
    fun deleteExpiredTokens(@Param("now") now: LocalDateTime?)
}
